posthog/hogql/ast.py

```diff
@@ -40,6 +41,21 @@ class VariableDeclaration(Declaration):
     expr: Optional[Expr] = None
 
 
+@dataclass(kw_only=True)
+class Block(Expr):
+    declarations: list[Declaration]
+    type: Optional[Type] = None
+
+    @cached_property
+    def placeholder_chain(self) -> str | None:
```

Comment:
I'd return a `list[str | int]` like a normal `ast.Field(...).chain` - feels like this would be more aligned